plugins/libplugin-pay.c

```diff
@@ -8,6 +8,8 @@
 
 #define DEFAULT_FINAL_CLTV_DELTA 9
 
+static u64 id_alloc = 0;
+
```

Comment:
FWIW this could also be done as a `static u64 next_id = 0` in the `payment_new` function. It's a static variable, initialized to 0 only the first time, and can be incremented from there. That'd avoid having a global variable here, and it'd be clearer where it is being used.